Libraries/src/Amazon.Lambda.Annotations.SourceGenerator/Validation/RouteParametersValidator.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Amazon.Lambda.Annotations.SourceGenerator.Models;
+using Amazon.Lambda.Annotations.SourceGenerator.Models.Attributes;
+
+namespace Amazon.Lambda.Annotations.SourceGenerator.Validation
+{
+    public static class RouteParametersValidator
+    {
+        private static readonly HashSet<string> NonFromRouteAttributes = new HashSet<string>
+        {
+            TypeFullNames.FromHeaderAttribute,
+            TypeFullNames.FromQueryAttribute,
+            TypeFullNames.FromBodyAttribute,
+            TypeFullNames.FromServiceAttribute
+        };
+
+        public static (bool isValid, IList<string> missingParameters) Validate(HashSet<string> routeParameters, IList<ParameterModel> lambdaMethodParameters)
+        {
+            var missingRouteParams = new List<string>();
+            var isValid = true;
+
+            foreach (var routeParam in routeParameters)
+            {
+                var fromRouteMethodParam = lambdaMethodParameters
+                    .FirstOrDefault(p => p.Attributes.Any(att => (att as AttributeModel<FromRouteAttribute>)?.Data.Name == routeParam));
+
+                // There exists a parameter attributed with FromRouteAttribute
```

Comment:
This comment is written as a partial thought. The comment should explain why it is worth skipping if we have a `FromRouteAttribute`. Maybe just rewording to something like "Route parameter valid because there is a matching signature parameter with a FromRoute attribute.

Reply:
Reworded.